src/DateRangePicker.ts

```diff
@@ -1,26 +1,423 @@
+
+    this.currentSlider = e.currentTarget as SVGRectElement;
+    const sliderX =
+      (this.currentSlider.id as SliderIds) === 'slider-min'
+        ? this._leftSliderX
+        : this._rightSliderX;
+    this._dragOffset = e.offsetX - sliderX;
+    this.currentSlider.classList.add('dragging');
+    this.histogram.addEventListener('pointermove', this.move);
+
+    // ensure that we catch the pointer up event, no matter where the pointer is
+    // when it's released. for more info see
+    // https://developers.google.com/web/updates/2016/10/pointer-events#pointer_capture
+    if (e.pointerId) {
+      this.histogram.setPointerCapture(e.pointerId);
+    }
   }
 
-  render() {
+  drop(): void {
+    this.histogram.removeEventListener('pointermove', this.move);
+    if (this.currentSlider) {
+      this.currentSlider.classList.remove('dragging');
+    }
+  }
+
+  // use an arrow function so that `this` is bound to the DateRangePicker object
+  // and not the event target. for more info see
+  // https://lit-element.polymer-project.org/guide/events#using-this-in-event-listeners
+  move = (e: PointerEvent): void => {
+    const newX = e.offsetX - this._dragOffset;
+    const slider = this.currentSlider as SVGRectElement;
+    return (slider.id as SliderIds) === 'slider-min'
+      ? this.setLeftSlider(newX)
+      : this.setRightSlider(newX);
+  };
+
+  setLeftSlider(newX: number): void {
+    const toSet = Math.min(newX, this._rightSliderX);
+    this._leftSliderX = Math.max(toSet, this.sliderWidth);
+  }
+
+  setRightSlider(newX: number): void {
+    const toSet = Math.max(newX, this._leftSliderX);
+    this._rightSliderX = Math.min(toSet, this.width - this.sliderWidth);
+  }
+
+  translatePositionToDate(x: number): string {
+    const milliseconds =
+      ((x - this.sliderWidth) * this.dateRange()) / this._histWidth;
+    const date = dayjs(this._minDate + milliseconds);
+    return date.format(this.dateFormat);
+  }
+
+  translateDateToPosition(date: string): number {
+    const milliseconds: number | undefined = dayjs(date).valueOf();
+    if (!milliseconds) {
+      return DATE_PARSING_ERROR;
+    }
+    // translate where we are within the date range into what the new x-position
+    // of the slider should be
+    return (
+      this.sliderWidth +
+      ((milliseconds - this._minDate) * this._histWidth) / this.dateRange()
+    );
+  }
+
+  handleDateInput(e: InputEvent): void {
+    const target = e.currentTarget as HTMLInputElement;
+    const newX = this.translateDateToPosition(target.value);
+    if ((target.id as InputIds) === 'date-min') {
+      if (newX !== DATE_PARSING_ERROR) {
+        this.setLeftSlider(newX);
+      }
+      target.value = this.translatePositionToDate(this._leftSliderX);
+    } else {
+      if (newX !== DATE_PARSING_ERROR) {
+        this.setRightSlider(newX);
+      }
+      target.value = this.translatePositionToDate(this._rightSliderX);
+    }
+  }
+
+  renderSlider(id: SliderIds): SVGTemplateResult {
```

Comment:
It looks like this method is doing a lot of conditionals to render the sliders. I might recommend making two higher-level methods for left and right sliders that configure the slider. ie:
```
private get leftSliderTemplate(): SVGTemplateResult {
  return this.generateSliderSVG(...);
}

private get rightSliderTemplate(): SVGTemplateResult {
  return this.generateSliderSVG(...);
}

private generateSliderSVG(x, k, c, sliderShape): SVGTemplateResult {
}

...

return html` ${this.leftSliderTemplate}  ${this.rightSliderTemplate} `;
```